### PR TITLE
Chore/add adapter doc

### PR DESCRIPTION
**Pull request by craigrbarnes**
Jira Ticket: [PXP-8657](https://ctds-planx.atlassian.net/browse/PXP-8657)

### Improvements
* Updated return types for metadate-adapters.
* Added documentation for the metadata-adapters.

### CODE REVIEW

File: src/mds/agg_mds/adapters.py [hunk 1]
```diff
@@ -139,6 +139,9 @@ def mapFields(item: dict, mappings: dict, global_filters: list = []) -> dict:
         :return:
         """
 
+        if global_filters is None:
+            global_filters = []
```
Comment (mfshao) [inline src/mds/agg_mds/adapters.py:143]: why this change?

**Review by mfshao — APPROVED** [review]

File: src/mds/agg_mds/adapters.py (hunk 1 above)
Comment (craigrbarnes) [reply src/mds/agg_mds/adapters.py:143]: Noticed that it was being flagged (PyCharm flagged it as well as pylint) as a Mutable Default Argument. See https://docs.python-guide.org/writing/gotchas/ for a description and resolution.

Comment (mfshao) [reply src/mds/agg_mds/adapters.py:143]: ah cool! thanks
